File: ClientLib/src/jspice/Str2et.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package jspice;

import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author owner
 */
public class Str2et {

   private static final Logger LOG = Logger.getLogger(Str2et.class.getName());
   private static final String COMMAND = "STR2ET";

   private String utc = null;
   private String status = null;
   private double et = Double.NaN;

   private Str2et() {
   }

   public Str2et(String fileName) {
      this.utc = fileName;
      this.status = null;
      this.et = Double.NaN;
   }

   public double getEt() {
      return et;
   }

   public String getUtc() {
      return utc;
   }

   public void setUtc(String utc) {
      this.utc = utc;
      this.status = null;
      this.et = Double.NaN;
   }

   @Override
   public String toString() {
      String s = Str2et.COMMAND;
      s = s + GetSocket.DELIMITER;
      s = s + this.getUtc();
      s = s + GetSocket.DELIMITER;
      return s;
   }

   public byte[] toBytes() {
      return this.toString().getBytes();
   }

   public String process() {
      LOG.log(Level.INFO, "process() sends = " + this.toString());
      GetSocket.sendReceive(this.toBytes());
      this.status = GetSocket.receiveToString();
      LOG.log(Level.INFO, "process() received = " + this.status);
      this.et = Double.parseDouble(this.status);
      return this.status;
   }

}

File: ServerLib/src/Str2et.c
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

#include "ProcessSpiceCommands.h"
#include "Str2et.h"
#include "SpiceUsr.h"

void Str2et(char* token, char* obuf) {

   printf("INFO: Str2et token = %s\n", token);
   token = strtok(NULL, DELIMITER);
   printf("INFO: Str2et received utc = %s\n", token);
   SpiceDouble et;
   str2et_c(token, &et);
   printf("INFO: Str2et str2et_c returned ephemeris time = %f \n", et);
   sprintf(obuf, "%f", et);
   printf("INFO: Str2et obuf = %s\n", obuf);
   printf("INFO: Str2et strlen(obuf) = %d\n", strlen(obuf));
  
}